servers/src/main/java/tachyon/master/block/BlockMaster.java

```diff
@@ -453,6 +453,18 @@ public long getWorkerId(NetAddress workerNetAddress) {
         return oldWorkerId;
       }
 
+      if (mLostWorkers.contains(mAddressIndex, workerAddress)) { // this is one of the lost workers
+        final MasterWorkerInfo lostWorkerInfo = mLostWorkers.getFirstByField(mAddressIndex,
+            workerAddress);
+        final long lostWorkerId = lostWorkerInfo.getId();
+        LOG.warn("A lost worker " + workerAddress + " has requested its old id "
+            + lostWorkerId + ".");
+
+        mWorkers.add(lostWorkerInfo);
+        mLostWorkers.remove(lostWorkerInfo);
```

Comment:
Yes, but this is taken into account when the worker, after calling `getWorkerId` will call `workerRegister`